src/components/AutoSuggest/index.js

```diff
@@ -0,0 +1,252 @@
+import React, { useCallback, useEffect, useState, useRef } from 'react';
+import PropTypes from 'prop-types';
+import styled from 'styled-components';
+import debounce from 'lodash/debounce';
+
+import useFetch from 'hooks/useFetch';
+import Input from 'components/Input';
+import SuggestList from './SuggestList';
+
+const formatFunc = ({ response }) => response.docs.map(({ id, weergavenaam }) => ({ id, value: weergavenaam }));
+const numOptionsFunc = data => data?.response?.docs?.length || 0;
+const INPUT_DELAY = 350;
+
+const Wrapper = styled.div`
+  position: relative;
+  z-index: 1;
+  min-width: 350px;
+`;
+
+const AbsoluteList = styled(SuggestList)`
+  position: absolute;
+  width: 100%;
+  background-color: white;
+`;
+
+// https://www.pdok.nl/restful-api/-/article/pdok-locatieserver#/paths/~1suggest/get
+const serviceParams = [
+  ['fq', 'bron:BAG'],
+  ['fq', 'type:adres'],
+  // ['fq', 'gemeentenaam:(amsterdam OR weesp)'],
+  ['fq', 'gemeentenaam:amsterdam'],
+  ['q', ''],
+];
+const asService = 'https://geodata.nationaalgeoregister.nl/locatieserver/v3/suggest?';
+const URL = `${asService}`.concat(serviceParams.flatMap(([key, value]) => `${key}=${value}`).join('&'));
+
+/**
+ * Autosuggest component that renders a text box and a list with suggestions after text input
+ *
+ * The text input is delayed by 350ms to prevent flooding of the REST service. Both rendered elements are fully
+ * accessibly and have corresponding ARIA 1.1 rules according to spec.
+ *
+ * Keyboard navigation is handled in the component where (when the list is open):
+ * - Tab focuses or blurs the text box, but not the list options
+ * - Up and down keys cycle through the list of options, keeping focus on the input field
+ * - Escape closes the list, focuses and clears the input field
+ * - Home key focuses the input field at the first character
+ * - End key focuses the input field at the last character
+ */
+const AutoSuggest = ({
+  className,
+  formatResponse = formatFunc,
+  numOptionsDeterminer = numOptionsFunc,
+  onSelect,
+  url = URL,
+}) => {
+  const { get, data } = useFetch();
+  const [show, setShow] = useState(false);
+  const [activeIndex, setActiveIndex] = useState(-1);
+  const wrapperRef = useRef(null);
+  const inputRef = useRef(null);
+
+  const handleKeyDown = useCallback(
+    event => {
+      const numberOfOptions = numOptionsDeterminer(data);
+      if (!show) return;
+
+      switch (event.keyCode) {
```

Comment:
Why would you not use `key names` here instead of `keyCode`, `key names` would make it more readable IMHO?

See: https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/key

```js
window.addEventListener("keydown", function (event) {
  if (event.defaultPrevented) {
    return; // Do nothing if the event was already processed
  }

  switch (event.key) {
    case "Down": // IE/Edge specific value
    case "ArrowDown":
      // Do something for "down arrow" key press.
      break;
    case "Up": // IE/Edge specific value
    case "ArrowUp":
      // Do something for "up arrow" key press.
      break;
```

Reply:
Done

Reply:
Nice!